cvat/apps/engine/location.py

```diff
@@ -14,31 +14,25 @@ class StorageType(str, Enum):
     def __str__(self):
         return self.value
 
-def get_location_configuration(obj, field_name: str, use_settings: bool = False) -> Dict[str, Any]:
+def get_location_configuration(
+    obj: Union[Project, Task, Job, Dict],
+    field_name: str,
+    use_settings: bool = False,
+) -> Dict[str, Any]:
     location_conf = {
         "is_default": use_settings
     }
 
     if use_settings:
         storage = getattr(obj, field_name) if not isinstance(obj, Job) else getattr(obj.segment.task, field_name)
-        if storage is None:
-            location_conf['location'] = Location.LOCAL
-        else:
-            location_conf['location'] = storage.location
-            sid = storage.cloud_storage_id
-            if sid:
-                location_conf['storage_id'] = sid
+        location_conf.update({
+            'location': storage.location if storage is not None else Location.LOCAL,
+            **({'storage_id': storage.cloud_storage.id} if getattr(storage, 'cloud_storage', None) else {})
+        })
```

Comment:
This seems like a downgrade compared to the previous version of the code. It's now much harder to understand what's happening (and I don't think the logic is actually any different?).

Also, you should use `storage.cloud_storage_id` instead of `storage.cloud_storage.id`, as the latter will fetch the cloud storage record from the database, even through you don't need it here.

Reply:
Okay, I rolled back the previous version if you like it better.